src/Compilers/Core/Portable/CommandLine/CommonCommandLineArguments.cs

```diff
@@ -279,6 +279,40 @@ internal CommandLineArguments()
         {
         }
 
+        internal static string NormalizePathPrefix(string filePath, ImmutableArray<KeyValuePair<string, string>> pathMap)
```

Comment:
PathUtilities felt like a place which was general path manipulation free of any compiler feature logic.  

As said before though, I don't have a strong opinion here.  Just couldn't find a perfect fit.